File: army_classes/MyArmy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MiddleAges
{
    class MyArmy : Army
    {
        //#################################################
        // CONSTRUCTORS
        //#################################################
        public MyArmy() { }

        public MyArmy(int limit)
        {
            ArmyLimit = limit;
        }

        //#################################################
        // MEMBER FUNCTIONS
        //#################################################
        public void AddArmyLimit(int value)
        {
            _armyLimit += value;
        }

        public void Buy(int value, int quantity, int gold, int iron, int food)
        {
            AddQuantity(quantity);
            AddPower(value);
            IncreaseBuyPrice(gold, 0, 0, iron, food);
        }

        public void Upgrade(double powerRate, int gold, int food, int iron)
        {
            _level += 1;

            MultiplyPower(powerRate);

            switch (_level)
            {
                case 1:
                    IncreaseUpgradePrice(gold, 0, 0, iron, food);
                    break;
                case 2:
                    IncreaseUpgradePrice(1.1 * gold, 0, 0, 1.1 * iron, 1.1 * food);
                    break;
                case 3:
                    IncreaseUpgradePrice(1.2 * gold, 0, 0, 1.2 * iron, 1.2 * food);
                    break;
                case 4:
                    IncreaseUpgradePrice(1.3 * gold, 0, 0, 1.3 * iron, 1.3 * food);
                    break;
                case 5:
                    IncreaseUpgradePrice(1.4 * gold, 0, 0, 1.4 * iron, 1.4 * food);
                    break;
                default:
                    IncreaseUpgradePrice(0, 0, 0, 0, 0);
                    break;
            }
        }

        //#################################################
        // GETTERS & SETTERS
        //#################################################
        public int ArmyLimit
        {
            get { return _armyLimit; }
            set { _armyLimit = value; }
        }

        //#################################################
        // MEMBER VARIABLES
        //#################################################
        private int _armyLimit;
    }
}

File: Etap 4 - Implementacja C++/Implementacja_C++/MiddleAges/myarmy.cpp
#include "myarmy.h"

//#################################################
// CONSTRUCTORS
//#################################################
MyArmy::MyArmy()
{

}

MyArmy::MyArmy(int limit)
{
    setArmyLimit(limit);
}

//#################################################
// MEMBER FUNCTIONS
//#################################################
void MyArmy::addArmyLimit(int value)
{
    armyLimit += value;
}


void MyArmy::buy(int value, int quantity, int gold, int iron, int food)
{
    addQuantity(quantity);
    addPower(value);
    increaseBuyPrice(gold, 0, 0, iron, food);
}

void MyArmy::upgrade(double powerRate, int gold, int food, int iron)
{
    level += 1;

    multiplyPower(powerRate);

    switch (level)
    {
    case 1:
        increasePrice(gold, 0, 0, iron, food);
        break;
    case 2:
        increasePrice(1.1 * gold, 0, 0, 1.1 * iron, 1.1 * food);
        break;
    case 3:
        increasePrice(1.2 * gold, 0, 0, 1.2 * iron, 1.2 * food);
        break;
    case 4:
        increasePrice(1.3 * gold, 0, 0, 1.3 * iron, 1.3 * food);
        break;
    case 5:
        increasePrice(1.4 * gold, 0, 0, 1.4 * iron, 1.4 * food);
        break;
    default:
        increasePrice(0, 0, 0, 0, 0);
    }
}


//#################################################
// SETTERS
//#################################################
void MyArmy::setArmyLimit(int value)
{
    armyLimit = value;
}

//#################################################
// GETTERS
//#################################################
int MyArmy::getArmyLimit()
{
    return armyLimit;
}
